internal/configs/version1/nginx-plus.ingress.tmpl

```diff
@@ -1,7 +1,7 @@
 # configuration for {{.Ingress.Namespace}}/{{.Ingress.Name}}
 {{range $upstream := .Upstreams}}
 upstream {{$upstream.Name}} {
-	zone {{$upstream.Name}} 256k;
+	{{if ne $upstream.UpstreamZoneSize "0"}}zone {{$upstream.Name}} {{$upstream.UpstreamZoneSize}};{{end}}
```

Comment:
For NGINX Plus, we always need to use zones. In your implementation, when the zone in the configmap or annotation is 0, that will disable the zone for Plus.